extensions/amp-story/1.0/amp-story-open-page-attachment.js

```diff
@@ -152,12 +153,44 @@ const renderOutlinkPageAttachmentUI = (
     openAttachmentEl.setAttribute('href', attachmentHref);
   }
 
-  // Setting theme
-  openAttachmentEl.setAttribute('theme', attachmentEl.getAttribute('theme'));
-
   // Getting elements
   const {chipEl, ctaLabelEl} = htmlRefs(openAttachmentEl);
 
+  // Setting theme
+  const themeAttribute = attachmentEl.getAttribute('theme');
+  openAttachmentEl.setAttribute('theme', themeAttribute);
+  let labelTextColor = themeAttribute === 'dark' ? 'white' : 'black';
+
+  if (themeAttribute === 'custom') {
+    const accentColor = attachmentEl.getAttribute('cta-accent-color');
+    // Calculating contrast color (black or white) needed for outlink CTA UI.
+    let contrastColor = null;
+    if (accentColor) {
+      setImportantStyles(attachmentEl, {
+        'background-color': attachmentEl.getAttribute('cta-accent-color'),
+      });
+      const styles = computedStyle(win, attachmentEl);
+      const rgb = getRGBFromCssColorValue(styles['background-color']);
+      contrastColor = getTextColorForRGB(rgb);
+      setImportantStyles(attachmentEl, {
+        'background-color': '',
+      });
+    }
+    if (attachmentEl.getAttribute('cta-accent-element') === 'background') {
+      labelTextColor = contrastColor;
+      setImportantStyles(openAttachmentEl, {
+        '--i-amphtml-outlink-cta-background-color': accentColor,
+        '--i-amphtml-outlink-cta-text-color': contrastColor,
+      });
+    } else {
+      labelTextColor = accentColor;
+      setImportantStyles(openAttachmentEl, {
+        '--i-amphtml-outlink-cta-background-color': contrastColor,
+        '--i-amphtml-outlink-cta-text-color': accentColor,
+      });
+    }
+  }
```

Comment:
I think this entire block could be extracted into a helper method to handle the custom theme and accent colors

Reply:
Wrote a `setCustomThemeStyles(win, attachmentEl, openAttachmentEl)` method